src/analyses/variable-sensitivity/array_abstract_object.h

```diff
@@ -26,6 +26,22 @@ class array_abstract_objectt:public abstract_objectt
     const abstract_environmentt &environment,
     const namespacet &ns);
 
+  virtual abstract_object_pointert read(
+    const abstract_environmentt &env,
+    const exprt &specifier,
+    const namespacet &ns) const override;
+
+  virtual abstract_object_pointert write(
+    abstract_environmentt &environment,
+    const namespacet &ns,
+    const std::stack<exprt> stack,
+    const exprt &specifier,
+    const abstract_object_pointert value,
+    bool merging_write) const override;
+
+protected:
+  CLONE
+
   virtual abstract_object_pointert read_index(
```

Comment:
I think they still need to be protected because in some cases sub-classes of `array_abstract_objectt` explicitly call back to these methods (e.g. doing a write_index on a `constant_array_abstract_objectt` that is bottom, causes that class to explicitly call back to `array_abstract_objectt::write_index`. Making them non-virtual might be ok though.